learning_resources/etl/micromasters.py

```diff
@@ -0,0 +1,114 @@
+"""MicroMasters ETL"""
+
+import requests
+from django.conf import settings
+
+from learning_resources.constants import LearningResourceType, OfferedBy, PlatformType
+from learning_resources.etl.constants import COMMON_HEADERS, ETLSource
+from learning_resources.models import LearningResource
+
+OFFERED_BY = {"name": OfferedBy.mitx.value}
+READABLE_ID_PREFIX = "micromasters-program-"
+MITXONLINE_PREFIX = "course-v1:"
+DEDP = "/dedp/"
+
+
+def _get_platform(program_url: str) -> str:
+    """Get the correct platform for a program"""
+    if DEDP in program_url:
+        return PlatformType.mitxonline.value
+    return PlatformType.edx.value
+
+
+def _get_course_id(program_url: str, edx_key: str) -> str:
+    """Get the correct readable id for a course"""
+    if DEDP in program_url and not edx_key.startswith(MITXONLINE_PREFIX):
+        return f"{MITXONLINE_PREFIX}{edx_key}"
+    return edx_key
+
+
+def extract():
+    """Loads the MicroMasters catalog data"""  # noqa: D401
+    if settings.MICROMASTERS_CATALOG_API_URL:
+        return requests.get(  # noqa: S113
+            settings.MICROMASTERS_CATALOG_API_URL, headers={**COMMON_HEADERS}
+        ).json()
+    return []
+
+
+def _is_published(course_id: str) -> bool:
+    """Determine if the course should be considered published"""
+    existing_course = LearningResource.objects.filter(
+        readable_id=course_id,
+        resource_type=LearningResourceType.course.value,
+        etl_source__in=[ETLSource.mit_edx.value, ETLSource.mitxonline.value],
+    ).first()
+    if existing_course:
+        return existing_course.published
+    return False
+
+
+def _transform_image(micromasters_data: dict) -> dict:
+    """
+    Transforms an image into our normalized data structure
+
+    Args:
+        micromasters_data (dict): micromasters program/course/run data
+
+    Returns:
+        dict: normalized image data
+    """  # noqa: D401
```

Comment:
You should rewrite this in imperative mood instead of using the noqa

Reply:
Probably the rest of the noqa codes in this file should be fixed and removed too - they were added to course_catalog to avoid rewriting code that is not going to be used anyway